p2p/src/connection/outgoing/p2p_connection_outgoing_reducer.rs

```diff
@@ -109,6 +105,7 @@ impl P2pConnectionOutgoingState {
                         time,
                         opts: opts.clone(),
                         rpc_id,
+                        on_success: None,
```

Comment:
what does it mean for it to be `None` here? why do we care about success in other instances but not during reconnection?

Reply:
`None` is here because we only need callback if we open outgoing connection from `rpc` layer or `kademlia` and need to update those layers after peer is connected

Reply:
Ok, maybe a dumb question, but would it make sense to differentiate between `Init` and `Reconnecting` state then? only `Init` would have the callback. The fact that one requires a callback and the other doesn't makes me think they are not really the same state then.